freqtrade/optimize/hyperopt/hyperopt_optimizer.py

```diff
@@ -377,33 +387,73 @@ def _get_results_dict(
+                    original_dim.low,
+                    original_dim.high,
+                    log=False,
+                )
+            elif isinstance(original_dim, Categorical):
+                o_dimensions[original_dim.name] = optuna.distributions.CategoricalDistribution(
+                    list(original_dim.bounds)
+                )
+            else:
+                raise Exception(f"Unknown search space {original_dim} / {type(original_dim)}")
+            # logger.info(f"convert_dimensions_to_optuna_space: {s_dimensions} - {o_dimensions}")
+        return o_dimensions
+
     def get_optimizer(
         self,
-        cpu_count: int,
         random_state: int,
-        initial_points: int,
-        model_queue_size: int,
-    ) -> Optimizer:
-        dimensions = self.dimensions
-        estimator = self.custom_hyperopt.generate_estimator(dimensions=dimensions)
-
-        acq_optimizer = "sampling"
-        if isinstance(estimator, str):
-            if estimator not in ("GP", "RF", "ET", "GBRT"):
-                raise OperationalException(f"Estimator {estimator} not supported.")
-            else:
-                acq_optimizer = "auto"
-
-        logger.info(f"Using estimator {estimator}.")
-        return Optimizer(
-            dimensions,
-            base_estimator=estimator,
-            acq_optimizer=acq_optimizer,
-            n_initial_points=initial_points,
-            acq_optimizer_kwargs={"n_jobs": cpu_count},
-            random_state=random_state,
-            model_queue_size=model_queue_size,
-        )
+    ):
+        o_sampler = self.custom_hyperopt.generate_estimator(dimensions=self.dimensions)
+        self.o_dimensions = self.convert_dimensions_to_optuna_space(self.dimensions)
+
+        # for save/restore
+        # with open("sampler.pkl", "wb") as fout:
+        #     pickle.dump(study.sampler, fout)
+        # restored_sampler = pickle.load(open("sampler.pkl", "rb"))
+
+        if isinstance(o_sampler, str):
+            if o_sampler not in (
+                "TPESampler",
+                "GPSampler",
+                "CmaEsSampler",
+                "NSGAIISampler",
+                "NSGAIIISampler",
+                "QMCSampler",
+            ):
+                raise OperationalException(f"Optuna Sampler {o_sampler} not supported.")
+
+        if o_sampler == "TPESampler":
+            sampler = optuna.samplers.TPESampler(seed=random_state)
+        elif o_sampler == "GPSampler":
+            sampler = optuna.samplers.GPSampler(seed=random_state)
+        elif o_sampler == "CmaEsSampler":
+            sampler = optuna.samplers.CmaEsSampler(seed=random_state)
+        elif o_sampler == "NSGAIISampler":
+            sampler = optuna.samplers.NSGAIISampler(seed=random_state)
+        elif o_sampler == "NSGAIIISampler":
+            sampler = optuna.samplers.NSGAIIISampler(seed=random_state)
+        elif o_sampler == "QMCSampler":
+            sampler = optuna.samplers.QMCSampler(seed=random_state)
```

Comment:
I'd prefer to keep the "override base estimator" functionality ([docs](https://www.freqtrade.io/en/latest/advanced-hyperopt/#overriding-base-estimator)) - but have that updated to support Optuna (compatible?) samplers instead (ignore compatibility for this functionality - but we'll need to update this part of the docs).

that will need some modifications to this - as currently you assume that o_sampler is a string (also after the "is_str" check).

As a consequence, i'd
* move the whole "if/elif" into the if block
* set "sampler = o_sampler" in the else block of the `isinstance` check
see if the "if/elif" can be simplified (dics often work well for this, especially considering that all parameters are identical.